numpy/random/mtrand/mtrand.pyx

```diff
@@ -376,28 +365,22 @@ cdef object discnp_array(rk_state *state, rk_discnp func, object size,
     cdef broadcast multi
 
     if size is None:
-        multi = <broadcast> PyArray_MultiIterNew(2, <void *>on, <void *>op)
-        array = <ndarray> PyArray_SimpleNew(multi.nd, multi.dimensions, NPY_LONG)
-        array_data = <long *>PyArray_DATA(array)
-        with lock, nogil:
-            for i from 0 <= i < multi.size:
-                on_data = <long *>PyArray_MultiIter_DATA(multi, 0)
-                op_data = <double *>PyArray_MultiIter_DATA(multi, 1)
-                array_data[i] = func(state, on_data[0], op_data[0])
-                PyArray_MultiIter_NEXT(multi)
+        multi = <broadcast>np.broadcast(on, op)
+        array = <ndarray>np.empty(multi.shape, dtype=int)
```

Comment:
I was following what had been done in the ```else``` block, but now I suddenly see that there is a gap in the testing there because the ```test_binomial``` function (and I suspect other functions as well) never tests broadcasting and hence the ```else``` block is not reached in the tests.

Is there perhaps a safer ```dtype``` we could use?  Perhaps ```np.long```?